internal/tmpl/tmpl.go

```diff
@@ -198,21 +206,16 @@ func (t *Template) WithBuildOptions(opts build.Options) *Template {
 }
 
 func buildOptsToFields(opts build.Options) Fields {
-	return Fields{
-		target:  opts.Target,
-		ext:     opts.Ext,
-		name:    opts.Name,
-		path:    opts.Path,
-		osKey:   opts.Goos,
-		arch:    opts.Goarch,
-		amd64:   opts.Goamd64,
-		go386:   opts.Go386,
-		arm:     opts.Goarm,
-		arm64:   opts.Goarm64,
-		mips:    opts.Gomips,
-		ppc64:   opts.Goppc64,
-		riscv64: opts.Goriscv64,
+	f := Fields{
+		target: opts.Target.String(),
+		ext:    opts.Ext,
+		name:   opts.Name,
+		path:   opts.Path,
```

Comment:
should probably still set all the keys from before, but empty, so it doesn't break anyone... might help template reuse...